src/ClusterBootstrap/cloud_init_aztools.py

```diff
@@ -565,6 +566,91 @@ def get_deployed_cluster_info(config, args):
         yaml.safe_dump({"machines": brief}, wf)
 
 
+def whitelist_source_address_prefixes():
+    resource_group = config["azure_cluster"]["resource_group"]
+    nsg_name = config["azure_cluster"]["nsg_name"]
+
+    cmd = """
+        az network nsg rule list \
+            --resource-group %s \
+            --nsg-name %s
+        """ % (resource_group,
+               nsg_name)
+
+    output = utils.exec_cmd_local(cmd)
+
+    try:
+        rules = json.loads(output)
+        for rule in rules:
+            if rule.get("name") == "whitelist":
+                return rule.get("sourceAddressPrefixes", [])
+    except Exception as e:
+        print("Exception: %s" % e)
+
+    return []
+
+
+def add_nsg_rule_whitelist(ips, dry_run=False):
```

Comment:
is it possible to encounter overlapping ranges? safer to use utils.keep_widest_subnet to pre-process, unless we can assume that the whitelist args are always valid.
also, users might need to know that input IPs are concatenated by comma